### PR TITLE
[SRVOCF-570] Add the `kn func subscribe` docs

### PR DESCRIPTION
**Pull request by rh-max**
Version(s):
`serverless-docs-1.32` +

Issue:
https://issues.redhat.com/browse/SRVOCF-570

Link to docs preview:
https://71546--docspreview.netlify.app/openshift-serverless/latest/functions/serverless-functions-subscribing

QE review:
- [ ] QE has approved this change.

### CODE REVIEW

**Comment by rh-max** [comment]
@matzew Could you PTAL? Thanks.

File: modules/serverless-subscribing-a-function-to-cloudevents.adoc [hunk 1]
```diff
@@ -0,0 +1,47 @@
+:_content-type: PROCEDURE
+[id="serverless-subscribing-a-function-to-cloudevents_{context}"]
+= Subscribing a function to CloudEvents
+
+The `subscribe` command connects the function to a set of events, matching a series of filters for `CloudEvent` metadata and a Knative Broker as the source of events, from where they are consumed.
+
+.Prerequisites
+
+* Knative Eventing is installed on the cluster.
+* You have installed the Knative (`kn`) CLI.
```
Comment (jrangelramos) [inline modules/serverless-subscribing-a-function-to-cloudevents.adoc:10]: Beyond knative eventing I think a configured broker is also required

File: modules/serverless-subscribing-a-function-to-cloudevents.adoc [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+.Procedure
+
+. Subscribe the function to events for a given broker:
++
+.Example command
+[source,terminal]
+----
+$ kn func subscribe --filter type=com.example --filter extension=my-extension-value --source my-broker
```
Comment (jrangelramos) [inline modules/serverless-subscribing-a-function-to-cloudevents.adoc:19]: I would probably go with a simpler sample with just one filter.. i.e:
```
kn func subscribe --filter type=com.example.Hello --source my-broker
```

File: modules/serverless-subscribing-a-function-to-cloudevents.adoc [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+$ kn func deploy
+----
++
+.Example output
+[source,terminal]
+----
+🙌 Function image built: <registry>/hello:latest
+🎯 Creating Triggers on the cluster
+✅ Function deployed in namespace "default" and exposed at URL:
+http://hello.default.127.0.0.1.example.com
```
Comment (jrangelramos) [inline modules/serverless-subscribing-a-function-to-cloudevents.adoc:46]: I think here we can omit the local address (on real cluster it won't appear).. 
```suggestion
http://hello.default.my-cluster.example.com
```

File: functions/serverless-functions-subscribing.adoc [hunk 4]
```diff
@@ -0,0 +1,11 @@
+:_content-type: ASSEMBLY
+[id="serverless-functions-subscribing"]
+= Subscribing functions to CloudEvents
+include::_attributes/common-attributes.adoc[]
+:context: serverless-functions-subscribing
+
+toc::[]
+
+You can subscribe a function to a set of events, which enables you to automate responding to them.
```
Comment (jrangelramos) [inline functions/serverless-functions-subscribing.adoc:9]: Hello, I know this description comes from the command help text, but perhaps we could re-word it to give a better description to user. But to be honest I dont have a better idea/description now.

**Comment by aireilly** [comment]
This pull request has been repointed to target serverless-docs-main. serverless-docs is no longer in use.

File: functions/serverless-functions-subscribing.adoc (hunk 4 above)
Comment (rh-max) [reply functions/serverless-functions-subscribing.adoc]: @jrangelramos Hi Jeff, does this look good?
`You can subscribe a function to a set of events. This links your function to `CloudEvent` objects defined by your filters and enables automated responses.`

File: modules/serverless-subscribing-a-function-to-cloudevents.adoc [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+.Prerequisites
+
+* You have installed Knative Eventing on the cluster.
+* You have configured a Knative Broker.
+* You have installed the Knative (`kn`) CLI.
+
+.Procedure
+
+. Subscribe the function to events for a given broker:
```
Comment (adellape) [inline modules/serverless-subscribing-a-function-to-cloudevents.adoc:15]: Consider the following list for [Procedure step that introduces a command](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#procedure-step-that-introduces-a-command), though it looks like the rest of the book doesn't really adhere either. 🤷 So maybe something for a future sweep.

**Review by adellape — APPROVED** [review]

File: functions/serverless-functions-subscribing.adoc (hunk 4 above)
Comment (jrangelramos) [reply functions/serverless-functions-subscribing.adoc]: Yea.. sounds a little better to me!

**Comment by rh-max** [comment]
/label merge-review-needed
To the merge reviewer: please do not merge on completion, we are waiting for the GA confirmation. Thank you.

**Review by adellape — APPROVED** [review]
Confirmed with Max via Slack that this is ready for merge.

**Pull request merged**